main.go

```diff
@@ -141,15 +144,27 @@ func ansibleEnable() {
 }
 
 func getAnsibleRepository(runDir string) error {
-	remoteURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), viper.GetString("http-url"))
+	httpURL := viper.GetString("http-url")
+	s3Obj := viper.GetString("s3-obj")
 	localCacheFile := fmt.Sprintf("/tmp/%s.tgz", appName)
 
-	err := idempotentFileDownload(remoteURL, localCacheFile, viper.GetString("http-user"), viper.GetString("http-pass"))
-	if err != nil {
-		return errors.Wrap(err, "unable to pull Ansible repo")
+	// Exactly one variable is defined
+	if (httpURL == "") == (s3Obj == "") {
+		return errors.New("exactly one remote resource must be specified. Choose one 'http-url' or 's3-obj'")
+	} else if httpURL != "" {
+		remoteHTTPURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), httpURL)
```

Comment:
Casing is a bit odd. `remoteHttpURL` seems better to me. But not a big deal.

Reply:
Done!